lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -340,6 +340,10 @@ def add_3ds(post, options)
           post[:browserInfo] = { userAgent: options[:user_agent], acceptHeader: options[:accept_header] }
           post[:additionalData] = { executeThreeD: 'true' } if options[:execute_threed]
         end
+        if options[:sca_exemption] && !options[:execute_threed].nil?
```

Comment:
This logic has some subtleties that might be worth locking down in the unit test suite. Could we add a couple tests which assert the expected shape of the outgoing request via a `check_request` block (a la `test_update_shopper_statement_and_industry_usage_sent`) given various combinations of `:sca_exemption` and `:execute_threed` options?

Reply:
This can be simplified by using `Hash#has_key?` eg:

```rb
if options.has_key?(:sca_exception) && options.has_key?(:execute_threed)
  ...
end
```

Reply:
Good call 👍 Since from what i've seen, these exemptions are only relevant for 3ds2, moved this snippet of logic up to the conditional related to 3ds2 and added unit tests to check that values are sent as expected